src/course-unit/course-xblock/CourseXBlock.jsx

```diff
@@ -0,0 +1,104 @@
+} from '@openedx/paragon';
+import { EditOutline as EditIcon, MoreVert as MoveVertIcon } from '@openedx/paragon/icons';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import { scrollToElement } from '../../course-outline/utils';
+import messages from './messages';
+
+const CourseXBlock = ({
+  id, title, unitXBlockActions, shouldScroll, ...props
+}) => {
+  const courseXBlockElementRef = useRef(null);
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
+  const intl = useIntl();
+
+  const onXBlockDelete = () => {
+    unitXBlockActions.handleDelete(id);
+    closeDeleteModal();
+  };
+
+  useEffect(() => {
+    // if this item has been newly added, scroll to it.
+    if (courseXBlockElementRef.current && shouldScroll) {
+      scrollToElement(courseXBlockElementRef.current);
+    }
+  }, []);
+
+  return (
+    <div ref={courseXBlockElementRef} {...props}>
+      <Card className="mb-1">
+        <Card.Header
+          title={title}
+          actions={(
+            <ActionRow>
+              <IconButton
+                alt={intl.formatMessage(messages.blockAltButtonEdit)}
+                iconAs={EditIcon}
+                size="md"
+                onClick={() => {}}
+              />
+              <Dropdown>
+                <Dropdown.Toggle
+                  id={id}
+                  as={IconButton}
+                  src={MoveVertIcon}
+                  alt={intl.formatMessage(messages.blockActionsDropdownAlt)}
+                  size="sm"
+                  iconAs={Icon}
+                />
+                <Dropdown.Menu>
+                  <Dropdown.Item>
+                    {intl.formatMessage(messages.blockLabelButtonCopy)}
+                  </Dropdown.Item>
+                  <Dropdown.Item onClick={() => unitXBlockActions.handleDuplicate(id)}>
+                    {intl.formatMessage(messages.blockLabelButtonDuplicate)}
+                  </Dropdown.Item>
+                  <Dropdown.Item>
+                    {intl.formatMessage(messages.blockLabelButtonMove)}
+                  </Dropdown.Item>
+                  <Dropdown.Item>
+                    {intl.formatMessage(messages.blockLabelButtonManageAccess)}
+                  </Dropdown.Item>
+                  <Dropdown.Item onClick={openDeleteModal}>
+                    {intl.formatMessage(messages.blockLabelButtonDelete)}
+                  </Dropdown.Item>
+                </Dropdown.Menu>
+              </Dropdown>
+              <DeleteModal
+                category="component"
+                isOpen={isDeleteModalOpen}
+                close={closeDeleteModal}
+                onDeleteSubmit={onXBlockDelete}
+              />
+            </ActionRow>
+          )}
+          size="md"
+        />
+        <Card.Section>
+          <div className="w-100 bg-gray-100" style={{ height: 200 }} data-block-id={id} />
```

Comment:
> `bg-gray-100`

Again, it seems to me that we're hard-coding an assumption about which color to use, where we could be more semantic instead.  In this case it's actually themable, but a little weird if you override a gray to be a brown.

Not a blocker, though.  Something to keep in mind going forward.

Reply:
Yes, I agree with you.
At this stage, the current implementation of the `CourseXBlock` component is not final. Ahead of us is the display of xblock content.